### DISCUSSION TITLE
NGINX sometimes keeps switching between all php versions installed (apache does not)

### DISCUSSION THREAD

**Discussion by IA21** [leokhoa/laragon#524] [General]
So I have installed 2 php versions (7.2.24 and 8.2.7) and everything is working perfectly. When I select any of them, I can check from phpinfo running on both apache (port 80) and nginx (port 8080). But sometimes, I've noticed nginx gets into this weird mode where it will cycle the php versions on reloads. 

e.g. When I'll load a file that calls `phpversion()` on apache, it will keep showing me 7.2.24 on all reloads since that's the one I have selected from laragon. But nginx will first show 7.2.24 and then ill reload and it will show 8.2.7 and then 7.2.24 again and so on. It only happens sometimes, I will reload the nginx service from laragon gui and it will stop happening. 

This is bit worrying because I don't know when nginx will just use the wrong php version. I am on Windows 10 64 bit, laragon and server versions are in the attached pic.

![image](https://github.com/leokhoa/laragon/assets/18244928/37b40176-4195-433c-a6d9-5df668397fc1)

**Comment by IA21** [comment]
ok I think its happening when I switch php versions and don't reload the servers afterwards. apache has no problem, nginx starts the cycling issue. But only sometimes, this cannot always reproduce the issue. 

This is worrying. For now I can tell nginx is causing this issue because i have phpversions right in front of me but when I start developing on it, I won't be able to tell and the version switches could cause bugs.

**Reply by Pen-y-Fan** [reply]
In Laragon 6 a **reload** doesn't always work with PHP 8, to be sure to load the correct version, after switching PHP, **Stop** and then **Start** Laragon.

**Reply by IA21** [reply]
I see. Is there a fix being worked on?

**Comment by Blackwhite779** [comment]
First, we will add PPA using the command below – ...
Run the Update command. ...
Install PHP 7.2 with this command. ...
Install PHP extensions using this command – ...
Now, install PHP 7.1 using this command – ...
Install common PHP7. ...
Install Nginx with this command –

**Comment by IA21** [comment]
> First, we will add PPA using the command below – ... Run the Update command. ... Install PHP 7.2 with this command. ... Install PHP extensions using this command – ... Now, install PHP 7.1 using this command – ... Install common PHP7. ... Install Nginx with this command –

I don't understand, what are you referring to?
